tools/qc/fastq_pair/fastq_pair.xml

```diff
@@ -0,0 +1,93 @@
+<tool id="fastq_pair" name="Paired-end fastq pairer" profile="18.01" version="1.0+galaxy0">
+    <description>Rewrite paired-end fastq files to make sure that all reads have a mate and to separate out singletons.</description>
+    <requirements>
+        <requirement type="package" version="1.0">fastq-pair</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+
+    ## Run fastq_pair
+    fastq_pair
+
+    #if $table_size:
+        -t '$table_size'
+    #end if
+
+    #if $print_elements:
+        -p '$print_elements'
+    #end if
+
+    #if $verbose:
+        '$verbose'
+    #end if
+
+    '$fastq1' '$fastq2' &&
+
+    #set fq1paired=$fastq1+".paired.fq"
+    #set fq2paired=$fastq2+".paired.fq"
+    #set fq1single=$fastq1+".single.fq"
+    #set fq2single=$fastq2+".single.fq"
```

Comment:
These statements won't work: `$fastq1` and `$fastq2` are params of type `data`. Their content will *print* as a file path on the generated command line, but inside python code (lines starting with `#`) you will have to deal with the actual object, which is of type `DatasetFilenameWrapper` in this case. What you want is `str($fastq1`, etc.

Reply:
I see now why I get an error mentioning that type.  Thanks very much!